### DISCUSSION TITLE
Welcome to Geode Discussions!

### DISCUSSION THREAD

**Discussion by HJfod** [geode-sdk/sdk#69] [Announcements]
## 👋 Welcome!
This is the **Discussions Page** for Geode. Here you can talk about things like:
  * Ask questions about how Geode works and how to use it
  * Share ideas for Geode
  * Share ideas for mods, request mods to be ported, etc.
  * Share links to catgirl photos

To get started, make a post, or reply to an existing one.

**Comment by TechStudent10** [comment]
Nice!

**Comment by rice7th** [comment]
Does geode have an internal widget api like better edit does?

**Reply by HJfod** [reply]
what widget system is this referring to?

**Reply by rice7th** [reply]
an API to interact with GMD's GUI and create new windows with buttons, sliders, text and things like that...

Basically a GUI lib

**Reply by HJfod** [reply]
Yeah Geode comes with Cocos2d headers, and a bunch of other utilities like string IDs and automatic layouts for modifying existing scenes, and utilities for creating popups, layers, etc..

It's not much of a 'GUI' lib tho, since buttons and stuff like that are a central part of any modding framework to the point that not having them would make the framework not really even qualify as one

**Comment by Djspac3** [comment]
already asked [here](https://github.com/geode-sdk/geode/discussions/554) but where the safety info like privacy?

**Reply by Djspac3** [reply]
even what is safe i looked at the website already where is it tho
